app/models/node.rb

```diff
@@ -221,9 +221,9 @@ def find_and_assign_last_inspect_report
     end
   end
 
-  def prune_reports(cutoff)
+  def prune_reports(cutoff, condition, status)
     transaction do
-      old_report_ids = self.reports.where('reports.time < ?', cutoff).pluck(:id)
+      old_report_ids = self.reports.where("reports.time < \"#{cutoff}\" and reports.status #{condition} \"#{status}\"").pluck(:id)
```

Comment:
I missed this one, same change as in the other file:
`+                         .where("reports.time < ? AND reports.status #{condition} ?", cutoff, status)
`